lib/mix/lib/mix/scm/git.ex

```diff
@@ -120,6 +154,21 @@ defmodule Mix.SCM.Git do
     end
   end
 
+  defp init_sparse(opts) do
+    git!("init --quiet")
+    git!("remote add origin #{opts[:git]} --fetch")
+    git!("config core.sparsecheckout true")
+
+    sparse_info =
+      File.cwd!
+      |> Path.join(".git/info/sparse-checkout")
+
+    Keyword.get_values(opts, :sparse)
```

Comment:
I don't think we should allow multiple sparse because `dest` will point to only one of them.